src/lib/utils/detect-insights-client.ts

```diff
@@ -0,0 +1,6 @@
+export default function hasDetectedInsightsClient(): boolean {
```

Comment:
If we want to completely strip it out of the production bundle, we might have to only implement the function in development mode (within a `__DEV__` check).

Reply:
where would you add `if (__DEV__)` condition? inside this function?

Reply:
You can see how it's done for the [`warning`](https://github.com/algolia/instantsearch.js/blob/50f4af3006a44cd08dd99b3a72bd410340c2e48a/src/lib/utils/logger.ts) function for example.

Reply:
Make sense, thanks!

Reply:
We could even inline this check in the `warning`, this would make it easier to strip it out in production.

Reply:
I checked the production build and I can confirm for the current PR, the function `hasDetectedInsightsClient ` has been completely tree shaked, without needing to inline it or add a similar logic as `warning`.

This is smarter than I expected and ideal, as if someone starts using the function for anything else than dev warnings, it will be included in the bundle no effort added.

Reply:
That's great! That's because of [this Babel plugin](https://github.com/algolia/instantsearch.js/blob/develop/scripts/babel/wrap-warning-with-dev-check.js) that wraps the warning inside a `__DEV__` check, then Rollup sees it's unused and doesn't include it in the bundle.